src/ucp/core/ucp_request.c

```diff
@@ -382,15 +380,12 @@ void ucp_request_dt_invalidate(ucp_request_t *req, ucs_status_t status)
                UCP_ERR_HANDLING_MODE_NONE);
     ucs_assert(UCP_DT_IS_CONTIG(req->send.datatype));
 
-    /* Do not allow destroying UCP endpoint, since the memeroy invalidation
-     * depends on it */
-    ucp_ep_refcount_add(req->send.ep, invalidate);
-
     req->send.state.uct_comp.count  = 1;
     req->send.state.uct_comp.func   = ucp_request_mem_invalidate_completion;
     req->send.state.uct_comp.status = UCS_OK;
     req->status                     = status;
     invalidate_map                  = ucp_request_get_invalidation_map(req);
+    req->send.worker                = worker;
```

Comment:
can we save it another field in the request and not union with ep?
can set req->send.ep=NULL

Reply:
done